### PR TITLE
New API to send data from Node throught SCTP DataConsumer

### PR DESCRIPTION
**Pull request by jmillan**

### CODE REVIEW

File: worker/include/PayloadChannel/Notification.hpp [hunk 1]
```diff
@@ -23,6 +23,9 @@ namespace PayloadChannel
 	private:
 		static std::unordered_map<std::string, EventId> string2EventId;
 
+	public:
```
Comment (ibc) [inline worker/include/PayloadChannel/Notification.hpp:23]: please static methods above static members.

File: worker/include/PayloadChannel/Request.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	public:
+		enum class MethodId
+		{
+			DATA_CONSUMER_SEND
+		};
+
+	private:
+		static std::unordered_map<std::string, MethodId> string2MethodId;
+
+	public:
```
Comment (ibc) [inline worker/include/PayloadChannel/Request.hpp:25]: static methods above static members.

**Comment by ibc** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/PayloadChannel.js  3
         

Complexity decreasing per file
==============================
+ lib/DataConsumer.js  -2
         

Clones added
============
- worker/src/RTC/Transport.cpp  2
- worker/src/PayloadChannel/UnixStreamSocket.cpp  1
- worker/src/PayloadChannel/Request.cpp  3
- lib/Channel.js  3
- lib/PayloadChannel.js  3
- worker/src/Channel/Request.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/versatica/mediasoup/pullRequest?prid=5980552&bid=19610425)

**Comment by nazar-pc** [comment]
I've got to touch this in Rust port and don't quite understand what is the purpose of this.
Seems a bit confusing to be able to send something to consumer that is attached to some producer already.

Could you explain a bit, please?

**Comment by ibc** [comment]
All this stuff will be reconsidered eventually on mediasoup v4.

The thing is: 

- `dataProducer.send()`: sends a message to the `Router` and the `Router` forwards it to all its `DataConsumers`.
- If any of those `DataConsumers` fail to send the message (SCTP buffer full, etc) there is no way for the app to deal with it.
- By adding `dataConsumer.send()` we let the app control message delivery and buffer the `message` in Node.js land for later if `send()` rejects.

**Pull request merged**